numpy/core/tests/test_umath.py

```diff
@@ -789,6 +789,32 @@ def assert_complex_equal(x, y):
                 assert_complex_equal(np.power(zero, -p), cnan)
             assert_complex_equal(np.power(zero, -1+0.2j), cnan)
 
+    # Testing 0^{Non-zero} issue 18378
+    def test_zero_power_nonzero(self):
```

Comment:
this test should be merged with the latter because the latter also is related to testing powers of zeros and actually test some cases of the type nan